src/geom/elem.C

```diff
@@ -602,34 +606,34 @@ void Elem::find_point_neighbors(const Point & p,
                 {
                   if (current_neighbor->active())                // ... if it is active
                     {
-                      if (current_neighbor->contains_point(p))   // ... and touches p
+                      auto it = neighbor_set.lower_bound(current_neighbor);
+                      if ((it == neighbor_set.end() || *it != current_neighbor) &&
+                          current_neighbor->contains_point(p))   // ... don't have and touches p
                         {
-                          // Make sure we'll test it
-                          if (!neighbor_set.count(current_neighbor))
-                            next_untested_set.insert (current_neighbor);
-
-                          // And add it
-                          neighbor_set.insert (current_neighbor);
+                          // Add it and test it
+                          next_untested_set.insert(current_neighbor);
+                          neighbor_set.emplace_hint(it, current_neighbor);
                         }
                     }
 #ifdef LIBMESH_ENABLE_AMR
                   else                                 // ... the neighbor is *not* active,
                     {                                  // ... so add *all* neighboring
                                                        // active children that touch p
-                      std::vector<const Elem *> active_neighbor_children;
-
+                      active_neighbor_children.clear();
```

Comment:
Are compilers still not smart enough to optimize a container declared in a loop as well as a container cleared at the start of a loop?  Has anyone tested that recently?  I admit I've made these sorts of optimizations myself and IIRC years ago they were a big deal, but I'd hoped that by 2020 we'd be able to just write purer code and let a smarter default allocator handle things.

Reply:
Hmm. I definitely haven't. I'm fine with this either way.

Reply:
I'd say leave it in.  I'd love some new benchmarks, but in a conflict between even half-recollected ancient benchmarks and hope, my money's not on hope.